idc_index/index.py

```diff
@@ -444,6 +528,254 @@ def get_viewer_URL(
 
         return viewer_url
 
+    def _get_series_size_from_crdc_series_uuid(
+        self, crdc_series_instance_uuid: str
+    ) -> float:
+        """
+        Retrieves the size of a series from the index based on the given CRDC series instance UUID.
+        As the index does only contains aws series urls, there is no direct way to
+        get series size from a gcs url. However this function levarages the
+        fact that both gcs and aws urls share the same folder name which is
+        crdc series instance uuid.
+
+        Args:
+            crdc_series_instance_uuid (str): The UUID of the CRDC series instance.
+
+        Returns:
+            float: The size of the series in MB.
+        """
+        index = self.index
+        series_size_sql = f"""
+            SELECT
+                series_size_MB
+            FROM
+                index
+            WHERE
+                series_aws_url LIKE '%{crdc_series_instance_uuid}%'
+        """
+        return duckdb.query(series_size_sql).to_df().series_size_MB.iloc[0]
+
+    def _validate_update_manifest_and_get_download_size(
+        self, manifestFile, downloadDir
+    ) -> tuple[float, str, Path]:
+        """
+        Validates the manifest file by checking the URLs and their availability.
+        The function reads the manifest file line by line. For each line, it checks if
+        the URL is valid and accessible.
+        Uses the s5cmd to check the availability of the URLs in both AWS and GCP.
```

Comment:
Is this a good idea? For large manifests, this potentially can be a lengthy operation. Did you test how long it takes? We should think how to optimize this. If both the bucket and the UUID are valid (based on the index), we can rather safely assume the URL is correct, and could avoid the costly operation.

Reply:
After rewriting the logic, we will be using download_from_manifest only when explicitly called. Given that, I think it may be important to check if the manifest was properly generated. 
I validated the 65k series manifest from portal, and it took >30 min(and counting, will update when finished) to validate.

> If both the bucket and the UUID are valid (based on the index), we can rather safely assume the URL is correct, and could avoid the costly operation.

We do not have access to the bucket id directly. But we could simplify validation for aws URLs by simply matching the entire s3 URLs in the index. If not found, then we could do `s5cmd ls`, to see if there are in GCP. What do you think? or should we simply extract and check if crdc instance uuid is in aws_url (this step is already included in the current PR)?

Reply:
I have a ridiculous idea that may or not be understood. But putting it out there. All of the manifest validation or sync operations are currently checked one series. I think there is a way we could check as a list of series. Here are my thoughts while commuting on mbta:



Remove commented lines from the manifest file.

Query against as if it is a text file with one column, extract crdc instance uuid from both index and left join with index using crdc instance uuid, convert to a data frame and then check if there are any empty rows in the index's crdc column.


Similarly for the sync command as well.

Reply:
Yep, join is exactly what I had in mind when I referred to checking bucket and UUID based on index! :-)

Reply:
Great.  For GCP urls, I'm checking only if crdc uuid is present in the index. We can't verify the bucket name. I think it may be a reasonable assumption. The validation at the df level for 65k series cohort is <10s now, but the sync dry run took ~12 min, so a total of ~13 min before download starts